ElixrTalentConnect/Login/LoginViewController.swift

```diff
@@ -0,0 +1,162 @@
+            signInPromptLabel.isUserInteractionEnabled = true
+            signInPromptLabel.addGestureRecognizer(signInGesture)
+            
+        }
+        
+        /// Function to setup SignUp action using gesture
+        /// - Parameter gesture: Specific part in the UIlabel where tap gesture is added.
+        @objc func signUpAction(gesture:UITapGestureRecognizer) {
+            let termsRange = (userInfoPrompt as NSString).range(of: self.textToChangeColor)
+            signInPromptLabel.isUserInteractionEnabled = true
+            if gesture.didTapAttributedTextInLabel(label: signInPromptLabel, inRange: termsRange) {
+                print("tapped")
+                guard let rootView = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "SignUpViewController") as? SignUpViewController
+                else {
+                    return
+                }
+                navigationController?.pushViewController(rootView, animated: true)
+            }
+        }
+        /// This function is called when the keyboard is about to be displayed.
+        ///It checks the size of the keyboard using information from the notification.
+        /// - Parameter notification: notification on clicking the textfields.
+        @objc func keyboardWillShow(notification: NSNotification) {
+            if let keyboardSize = (notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue {
+                if self.view.frame.origin.y == 0 {
+                    self.view.frame.origin.y -= keyboardSize.height
+                }
+            }
+        }
+        
+        /// This implementation is commonly used to handle the keyboard covering the text input fields in a view, ensuring a smooth user experience.
+        /// - Parameter notification: notification on clicking the textfields.
+        @objc func keyboardWillHide(notification: NSNotification) {
+            if self.view.frame.origin.y != 0 {
+                self.view.frame.origin.y = 0
+            }
+        }
+        
+        /// Function  to perform  navigation to Signup viewcontroller.
+        func tapSignup() {
+            guard   let homeVc = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "LoginViewController") as? LoginViewController else{
+                return
+            }
+            navigationController?.pushViewController(homeVc, animated: true)
+        }
+        
+        /// Function to  perform navigaion to HomeviewController.
+        func navigateToHome() {
+            guard   let HomeView = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "HomeViewController") as? HomeViewController else {
+                return
+            }
+            self.navigationController?.pushViewController(HomeView, animated: true)
+        }
+        
+        /// Function to display alert on empty .
+        func showAlert(message:String){
```

Comment:
If this alert method is used for any specific purpose it would be better to name it accordingly or mention the purpose in the method documentation. `showAlert(message: )` will seem like a generic alert message.